dvc/external_repo.py

```diff
@@ -123,6 +124,22 @@ def _set_cache_dir(self):
 
         self.cache.local.cache_dir = cache_dir
 
+    def _set_relative_upstream(self):
+        # check if the URL is local and if a default relative-path
+        # remote is present, if so, make the URL relative to the
+        # original repo
+        if os.path.isdir(self.url):
+            remote_name = self.config["core"].get("remote")
+            if remote_name:
+                src_repo = Repo(self.url)
+                remote_cfg = self.config["remote"][remote_name]
+                old_remote_url = src_repo.config["remote"][remote_name]["url"]
```

Comment:
remote url might be a non-local one, in which case you shouldn't modify it.

Reply:
I would say we can just assign with no check:
```python
self.config["remote"][name]["url"] = src_repo.config["remote"][name]["url"]
```
Since url should be the same anyway. Name method smth like `._fix_local_upstream()`, comment on what's the intention if that is not enough.

Reply:
Should an url for a remote in the erepo ever be different from the url of the matching remote in the original repository?
Because if not, then an equality check should be enough -- assuming we want to avoid unnecessary mutations to the config, otherwise it can be assigned directly as you said.

Reply:
Something like:

```python
new_remote = self.config["remote"][remote_name]
old_remote = src_repo.config["remote"][remote_name]
if new_remote["url"] != old_remote["url"]:
    new_remote["url"] = old_remote["url"]
```

An additional check, in case an url in the new repo could be different from an url in the old repo is to check that `self.root_dir` is a prefix of the new url (which also ensures that it is local and relative).

Reply:
Great point @Suor !

> Should an url for a remote in the erepo ever be different from the url of the matching remote in the original repository?

Yes, when `_set_upstream` is applied, because it sets it to the original cache location. But those are in different branches of execution, so it doesn't matter and we can simply assign here.